### ISSUE TITLE
Works unreliably in Windows 10 pro

### ISSUE THREAD

**Issue by taismi** [sivel/speedtest-cli#329]
Python 3.5.2. Windows 10 professional 64 bit. Retrieving speedtest.net configuration... fails quite often. Never fails in Windows 7 pro 64-bit. These is no any waiting in Retrieving speedtest.net configuration... state. Error is printed very soon.

```
C:\test>python speedtest_cli.py
Retrieving speedtest.net configuration...
Traceback (most recent call last):
  File "speedtest_cli.py", line 796, in <module>
    main()
  File "speedtest_cli.py", line 790, in main
    speedtest()
  File "speedtest_cli.py", line 623, in speedtest
    config = getConfig()
  File "speedtest_cli.py", line 385, in getConfig
    configxml.append(uh.read(10240))
  File "C:\Program Files (x86)\Python35-32\lib\http\client.py", line 448, in read
    n = self.readinto(b)
  File "C:\Program Files (x86)\Python35-32\lib\http\client.py", line 488, in readinto
    n = self.fp.readinto(b)
  File "C:\Program Files (x86)\Python35-32\lib\socket.py", line 575, in readinto
    return self._sock.recv_into(b)
ConnectionResetError: [WinError 10054] 
```

**Comment by taismi** [comment]
Problem occurs in VMs.

**Comment by sivel** [comment]
Please give the devel branch a try and see if the changes made there already resolve your problem.

https://raw.githubusercontent.com/sivel/speedtest-cli/devel/speedtest.py

**Comment by sivel** [comment]
Please test with the recently released v1.0.0.

**Comment by taismi** [comment]
I tried speedtest 6 times and 2 of them failed. Windows 7 pro 64-bit is running in Virtual Bos.

python.exe speedtest_cli.py
Retrieving speedtest.net configuration...
Traceback (most recent call last):
  File "speedtest_cli.py", line 1410, in <module>
    main()
  File "speedtest_cli.py", line 1400, in main
    shell()
  File "speedtest_cli.py", line 1310, in shell
    speedtest = Speedtest()
  File "speedtest_cli.py", line 685, in __init__
    self.get_config()
  File "speedtest_cli.py", line 710, in get_config
    stream = get_response_stream(uh)
  File "speedtest_cli.py", line 417, in get_response_stream
    return GzipDecodedResponse(response)
  File "speedtest_cli.py", line 296, in __init__
    self.io = IO(response.read())
  File "c:\Ohjelmatiedostot\Python35 32\lib\http\client.py", line 461, in read
    s = self._safe_read(self.length)
  File "c:\Ohjelmatiedostot\Python35 32\lib\http\client.py", line 607, in _safe_read
    chunk = self.fp.read(min(amt, MAXAMOUNT))
  File "c:\Ohjelmatiedostot\Python35 32\lib\socket.py", line 575, in readinto
    return self._sock.recv_into(b)
ConnectionResetError: [WinError 10054]


python.exe speedtest_cli.py --version
1.0.0

**Comment by sivel** [comment]
This should be resolved in devel as part of 823d7dc2b7e9a7acc9769fb197eb9f56a36444a9
